Symfony/CS/Tokenizer/TokensAnalyzer.php

```diff
@@ -211,7 +211,79 @@ public function isArrayMultiLine($index)
     }
 
     /**
-     * Check if there is a lambda function under given index.
+     * Returns the attributes of the method under the given index.
+     *
+     * @param int $index Token index of the method (T_FUNCTION)
+     *
+     * @return array
+     */
+    public function getMethodAttributes($index)
+    {
+        $tokens = $this->tokens;
+        $token = $tokens[$index];
+
+        if (!$token->isGivenKind(T_FUNCTION)) {
```

Comment:
Good question, same goes for: if it is a function but not a method.

Throwing an exception would make sense, since the caller asks for attributes about a method but didn't provide a reference (index) to a method to gather information on. On the other hand: this function could be used to check if the given T_FUNCTION is a method (like we have for lambda).

We could also make this into a more generic function; given back information about the T_FUNCTION like `isLambda`, `isMethod` (or `isFunction`) as well.

How would like me to proceed here? (others options are welcome as ever :) )